src/components/Modals/HTMLContentModal/HTMLContentModal.js

```diff
@@ -18,228 +18,132 @@
-  flex: 1;
-  align-items: center;
-  justify-content: center;
-`;
-
-const boldStyle = { fontFamily: appFont.medium };
-
-const baseStyles = (colors) => {
-  return (
-    StyleSheet.create({
-      b: boldStyle,
-      strong: boldStyle,
-      a: {
-        ...boldStyle,
-        color: colors.basic000,
-        fontSize: fontSizes.regular,
-        lineHeight: lineHeights.regular,
-      },
-      li: { fontSize: fontSizes.regular, lineHeight: lineHeights.regular },
-      p: { fontSize: fontSizes.regular, lineHeight: lineHeights.regular },
-      h1: { ...boldStyle, fontSize: fontSizes.giant, lineHeight: lineHeights.giant },
-      h2: { ...boldStyle, fontSize: fontSizes.large, lineHeight: lineHeights.large },
-      h3: { ...boldStyle, fontSize: fontSizes.big, lineHeight: lineHeights.big },
-      h4: { ...boldStyle, fontSize: fontSizes.medium, lineHeight: lineHeights.medium },
-      h5: { ...boldStyle, fontSize: fontSizes.regular, lineHeight: lineHeights.regular },
-      h6: { ...boldStyle, fontSize: fontSizes.regular, lineHeight: lineHeights.regular },
-    })
-  );
-};
-
-class HTMLContentModal extends React.Component<Props, State> {
-  scrollViewRef = React.createRef<React.ElementRef<typeof ScrollView>>();
-  modalRef = React.createRef();
-
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      isHtmlFetched: false,
-      htmlData: '',
-      scrollOffset: 0,
-      containerHeight: 0,
-      contentContainerHeight: 0,
-    };
-  }
-
-  componentDidMount() {
-    const endpointPrefix = firebaseRemoteConfig.getString(REMOTE_CONFIG.LEGAL_HTML_ENDPOINT_PREFIX);
-    const { htmlEndpoint } = this.props;
-    // eslint-disable-next-line i18next/no-literal-string
-    const htmlEndpointFull = `${endpointPrefix}${htmlEndpoint}.html`;
-
-    fetch(htmlEndpointFull)
-      .then((resp) => resp.text())
-      .then((text) => {
-        this.setState({
-          isHtmlFetched: true,
-          htmlData: text.replace(/(\r\n\t|\n|\r\t)/gm, ''),
-        });
-      })
-      .catch((error => {
-        reportErrorLog(`HTMLContentModal: Failed to fetch ${htmlEndpointFull}`, { error });
-      }));
-  }
-
-  handleModalClose = () => {
-    if (this.modalRef.current) this.modalRef.current.close();
-  };
 
-  renderNode = (node: CustomNode) => {
-    if (node.name === 'iframe' || node.name === 'script') {
-      return null;
+const HTMLContentModal = ({ primisicDocumentId }: Props) => {
+  const scrollViewRef = React.useRef(null);
+  const modalRef = React.useRef();
+  const [isPrimisicDocumentFetched, setIsPrimisicDocumentFetched] = React.useState(false);
+  const [documentHTMLData, setDocumentHTMLData] = React.useState('');
+  const [scrollOffset, setScrollOffset] = React.useState(0);
+  const [containerHeight, setContainerHeight] = React.useState(0);
+  const [contentContainerHeight, setContentContainerHeight] = React.useState(0);
+
+  React.useEffect(() => {
+    async function fetchPrimisicData() {
```

Comment:
Can we move most of logic (fetching, mapping) to `services/prismic` and then only use the output for state here?

Reply:
Done.